### PR TITLE
refactor(cdr-tabs and cdr-tab-panel): rewrite cdr-tabs for vue 3

### PR DESCRIPTION
**Pull request by KenjiCrosland**
Almost a complete rewrite of the tabs component. Changed markup to be more semantically appropriate.
Most styling remains unchanged, with a few changes to account for the different markup. Most of the
same CSS classes are used. Logic dealing with animations and overflowgradient have been copy/pasted
from the old version of tabs and integrated into this new version. Some refactoring of the old tests
have been done, but it's far from a complete overhaul--just ensuring that all the existing tests
still test what they had done in the past.

### CODE REVIEW

File: src/components/tabs/CdrTabs.vue [hunk 1]
```diff
@@ -89,276 +23,244 @@ export default defineComponent({
...
+        const scrollX = tablist.value.scrollLeft;
         overflowLeft.value = scrollX > 1;
         overflowRight.value = (scrollX + 1) < (headerWidth.value - containerWidth);
-      } else {
+    } else {
         overflowLeft.value = false;
         overflowRight.value = false;
-      }
     }
-    const updateUnderline = () => {
```
Comment (KenjiCrosland) [inline src/components/tabs/CdrTabs.vue:233]: updateUnderline was migrated from the old version of tabs. I just renamed some variables to account for the change in markup.

File: src/components/tabs/CdrTabs.vue [hunk 2]
```diff
@@ -89,276 +23,244 @@ export default defineComponent({
...
-        }
-        totalWidth += element.offsetWidth || 0;
-      });
-      return totalWidth;
+    totalWidth += element.offsetWidth || 0;
+    });
+    return totalWidth;
+};
+
+const updateUnderline = () => {
```
Comment (KenjiCrosland) [inline src/components/tabs/CdrTabs.vue:102]: This is largely unchanged from its original version

File: src/components/tabs/CdrTabs.vue [hunk 3]
```diff
@@ -89,276 +23,244 @@ export default defineComponent({
...
+      <li v-for="(tab, index) in tabs" 
+        :key="tab.name" role="presentation"
+        :class="style['cdr-tabs__header']" 
+      >
+        <button
+          :ref="el => { tabElements[index] = el }"
+          :id="getTabId(tab.name)"
+          :disabled="tab.disabled"
+          :aria-disabled="tab.disabled"
+          :aria-selected="(selectedIndex === index && !tab.disabled) ? true : false"
```
Comment (KenjiCrosland) [inline src/components/tabs/CdrTabs.vue:232]: Is there a way to get a computed value for `(selectedIndex === index && !tab.disabled)`? It's in a `v-for` loop so it didn't seem that simple. Anyway it bothered me that I was repeating this 3 times.

**Review by mhewson — APPROVED** [review]

File: src/components/tabs/CdrTabs.vue [hunk 4]
```diff
@@ -228,12 +230,11 @@ const style = useCssModule();
           :ref="el => { tabElements[index] = el }"
           :id="getTabId(tab.name)"
           :disabled="tab.disabled"
-          :aria-disabled="tab.disabled"
-          :aria-selected="(selectedIndex === index && !tab.disabled) ? true : false"
-          :tabIndex="(selectedIndex === index && !tab.disabled) ? 0 : -1"
+          :aria-selected="checkIfActive(index, tab) ? true : false"
```
Comment (benjag) [inline src/components/tabs/CdrTabs.vue:233]: Isn't the ternary here redundant? You're already returning a true or false from the method

Comment (KenjiCrosland) [reply src/components/tabs/CdrTabs.vue]: Good catch. updated.

File: src/components/tabs/CdrTabPanel.vue [hunk 5]
```diff
@@ -1,117 +1,34 @@
...
-  <div
-    :aria-hidden="!active"
-    :aria-labelledby="ariaLabelledby"
-    :class="mapClasses(style, baseClass, modifierClass, animationDirection && `cdr-tab-panel-${animationDirection}`)"
-    :hidden="hidden"
-    :id="id"
+  <section
+    v-show="isActive"
     tabindex="0"
+    :aria-visible="isActive ? true : false"
```
Comment (benjag) [inline src/components/tabs/CdrTabPanel.vue:23]: redundant ternary

File: src/components/tabs/examples/Tabs.vue [hunk 6]
```diff
@@ -237,4 +237,4 @@ export default {
   .tab-demo-section {
     margin-bottom: 30px;
   }
-</style>
+</style>
```
Comment (benjag) [inline src/components/tabs/examples/Tabs.vue:240]: Note the 'no newline' lint errro

File: src/components/tabs/examples/demo/TabsDefault.vue [hunk 7]
```diff
@@ -174,4 +175,4 @@ export default {
   h3.tab-title {
     margin: 20px 0;
   }
-</style>
+</style>
```
Comment (benjag) [inline src/components/tabs/examples/demo/TabsDefault.vue:178]: newline

**Review by benjag — APPROVED** [review]

**Pull request merged**